api/client/client.go

```diff
@@ -3146,6 +3146,16 @@ func GetResourcePage[T types.ResourceWithLabels](ctx context.Context, clt GetRes
 		req.Limit = int32(defaults.DefaultChunkSize)
 	}
 
+	// DELETE IN: v15.0.0 (required to ensure that tsh's resource-based access request generation and ambiguous host error formatting
+	// behave correctly when talking to older v12/v13 auth servers).
+	var routeMatcher utils.SSHRouteMatcher
+	if req.SSHTargetHost != "" {
+		routeMatcher = utils.NewSSHRouteMatcher(
+			req.SSHTargetHost, req.SSHTargetPort,
+			utils.WithCaseInsensitiveRouting(true),
```

Comment:
Defaulting to `true` is imperfect, but defaulting to `false` would be worse as it would filter out case-insensitive matches even when case insensitive routing was turned on.  Generally client-side filtering should always select the more permissive option if things are tunable, since it's performing additional filtering _on top of_ the filtering already performed server-side.

Obviously the truly "correct" way to handle this would be to check the actual cluster networking config, but I don't want to add additional hidden network round-trips... I'll give this some more thought tho, I do feel like this could be better...

Reply:
Ended up scrapping this code and doing a separate method.  There is still a fallback with client-side filtering, but now we only hit it if we get a `NotImplemented` error, so we know that if we hit it `case_insensitive_routing` must be `false`.  Easy-peasy.